### DISCUSSION TITLE
Form encode a value of the field before submission

### DISCUSSION THREAD

**Discussion by janforward** [elementor/elementor#24795] [Developers Community]
### Start a discussion

We would like to encode a field from the form with sha1 before saving it to the database and adding it to submissions.
We tried many solutions but none of them actually worked and the values are always saved in to the database as original values (values that are inputed into the form by user). Our code which is not working:

```
add_action( 'elementor_pro/forms/process', function( $record, $ajax_handler ) {
    // Make sure we are working with the correct form
    $form_id = $record->get_form_settings( 'form_id' );
    if ( 'test' !== $form_id ) {
        return;
    }

    $raw_fields = $record->get( 'fields' );

    foreach ( $raw_fields as $id => $field ) {
        // Check if this is the field to encode
        if ( 'emso' === $id ) {
            $raw_fields[ $id ]['value'] = sha1( $field['value'] );
        }
    }

    // Update the form record with the encoded value
    $record->update_fields( $raw_fields );
}, 10, 2 );
```

I would really appreciate any help on this matter! :)

**Comment by perks98** [comment]
@janforward from checking the developer docs have you tried the hook `elementor_pro/forms/process/{$field_type}` as in the docs the hook you are using mentions "Fires after the forms fields have been validated and processed", while the one I mentioned in the docs says "Fires when a single form field is being processed". Not sure if that helps as I haven't personally used those hooks before

[https://developers.elementor.com/docs/hooks/forms/#elementor-pro-forms-process-field-type](https://developers.elementor.com/docs/hooks/forms/#elementor-pro-forms-process-field-type)

**Reply by janforward** [reply]
@perks98 thanks for help! I tried to update the code based on your suggestion, but unfortunatelly it didn't work for us:

```
add_action( 'elementor_pro/forms/process/text', function( $field, $record, $ajax_handler ) {
    // Make sure we are working with the correct form
    $form_id = $record->get_form_settings( 'form_id' );
    if ( 'test' !== $form_id ) {
        return;
    }

    // Check if this is the field to encode
    if ( 'emso' === $field['id'] ) {
        $field['value'] = sha1( $field['value'] );

        // Update the specific field with the encoded value
        $record->update_field( $field['id'], $field['value'] );
    }
}, 10, 3 );
```

**Reply by perks98** [reply]
@janforward I have just tried it out myself and got stuck like you did. But from checking Elementor Pro code it seems what was incorrect was how it updates the field, it is missing the 'value' parameter in your update_field function. Try the code below:

```
add_action( 'elementor_pro/forms/process/text', function( $field, $record, $ajax_handler ) {
    // Make sure we are working with the correct form
    $form_id = $record->get_form_settings( 'form_id' );
    if ( 'test' !== $form_id ) {
        return;
    }

    // Check if this is the field to encode
    if ( 'emso' === $field['id'] ) {
        // Update the specific field with the encoded value
        $record->update_field( $field['id'], 'value', sha1( $field['value']) );
    }
}, 10, 3 );
```

**Reply by janforward** [reply]
@perks98 Works as a charm :). Thank you so much!!
